### DISCUSSION TITLE
how to selected / old value ?

### DISCUSSION THREAD

**Discussion by bogordesaincom** [mantinedev/mantine#410] [Q&A] [answered]
How to selected old value with `Select Component`
[Select](https://mantine.dev/core/select/)

**Comment by rtivital** [comment, marked as answer]
I do not understand your question. Please make it more clear and provide a code example of what you are trying to achieve.

**Reply by bogordesaincom** [reply]
In CRUD where Edit Process 

how to load value role field?

<img width="603" alt="Jepretan Layar 2021-11-12 pukul 15 00 15" src="https://user-images.githubusercontent.com/51335197/141431581-ceb7bfc2-4a84-492b-8ad9-3571fa07bd32.png">


This my data list

<img width="624" alt="Jepretan Layar 2021-11-12 pukul 15 01 25" src="https://user-images.githubusercontent.com/51335197/141431713-9f662c00-e579-48c5-9a7d-d95096629004.png">
 
```
    const [selectvalue, setSelectValue] = useState({});
    const [role_list, setRoleList] = useState([]);

<Select
                        style={{ marginTop: 20 }}
                        label="Roles"
                        placeholder="Select Roles"
// How to default value with load from old value in here
                        value={selectvalue}
                        onChange={setSelectValue}
                        data={role_list}
                    />
```

**Reply by rtivital** [reply]
You just need to set value like so in your state: 

```tsx
const initialValue = getValueFromAPI();
const [state, setState] = useState(initialState);
<Select value={state} onChange={setState} />
```

**Reply by bogordesaincom** [reply]
i was try 

```
const [other, otherState] = useState()
    const [selectvalue, setSelectValue] = useState(otherState)

const dataload = async() => {
let { data} = await axios.get(`/roles/${id}`)

// response object is data.data because this data is array, and 

// output {id, name, display_name}
otherState({value: data.data.id, label: data.data.display_name})

}
```
but not work

**Reply by bogordesaincom** [reply]
> You just need to set value like so in your state:
> 
> ```tsx
> > const [state, setState] = useState(initialState);
> <Select value={state} onChange={setState} />
> ```


const initialValue = getValueFromAPI();
it's object output? or array?

**Reply by rtivital** [reply]
It's a string, that is included in select data. You need to extract this value from your api response into react state.

**Reply by bogordesaincom** [reply]
ahhh... ok, i think this object, thanks for help @rtivital
